saleor/payment/migrations/0039_transactionevent_currency.py

```diff
@@ -0,0 +1,25 @@
+from django.db import migrations
+from django.apps import apps as registry
+from django.db.models.signals import post_migrate
+
+from .tasks.saleor3_12 import set_default_currency_for_transaction_event_task
+
+
+def set_default_currency_for_transaction_event(apps, _schema_editor):
+    def on_migrations_complete(sender=None, **kwargs):
+        set_default_currency_for_transaction_event_task.delay()
```

Comment:
We shouldn't trigger this task if there are no transaction events in the database.

Reply:
Keep in mind that even if they don't exist in the moment of checking it doesn't mean that they will not be created in the time period between migration and calling the task.

Reply:
Note: in new approach, most task will not be executed in case of empty table (we expect empty tables in migration when setting up empty database).